server/services/permissionsService.test.ts

```diff
@@ -0,0 +1,785 @@
+
+        expect(permissions.visits).toEqual({
+          view: true,
+        })
+      })
+
+      it('should return view true if user has ReceptionUser role', async () => {
+        const user = { ...prisonUserMock, userRoles: [Role.GlobalSearch, Role.ReceptionUser] }
+        const prisoner = { ...PrisonerMockDataA, prisonId: 'LEI' }
+
+        const permissions = await service.getOverviewPermissions(user, prisoner)
+
+        expect(permissions.visits).toEqual({
+          view: true,
+        })
+      })
+    })
+
+    describe('incentives', () => {
+      it('should return view true if prisoner is in case load', async () => {
+        const permissions = await service.getOverviewPermissions(prisonUserMock, PrisonerMockDataA)
+
+        expect(permissions.incentives).toEqual({
+          view: true,
+        })
+      })
+
+      it('should return view true if user has GlobalSearch role', async () => {
+        const user = { ...prisonUserMock, userRoles: [Role.GlobalSearch] }
+        const prisoner = { ...PrisonerMockDataA, prisonId: 'LEI' }
+
+        const permissions = await service.getOverviewPermissions(user, prisoner)
+
+        expect(permissions.incentives).toEqual({
+          view: true,
+        })
+      })
+    })
+
+    describe('category', () => {
+      it('should return edit false if user doesnt have appropriate role', async () => {
+        const permissions = await service.getOverviewPermissions(prisonUserMock, PrisonerMockDataA)
+
+        expect(permissions.category).toEqual({
+          view: false,
+          edit: false,
+        })
+      })
+
+      it.each([
+        Role.CreateRecategorisation,
+        Role.ApproveCategorisation,
+        Role.CreateRecategorisation,
+        Role.CategorisationSecurity,
+      ])('should return edit true if user has %s role', async role => {
+        const user = { ...prisonUserMock, userRoles: [role] }
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.category).toEqual({
+          view: false,
+          edit: true,
+        })
+      })
+    })
+
+    describe('calculateReleaseDates', () => {
+      it('should return edit false if user doesnt have appropriate role', async () => {
+        const permissions = await service.getOverviewPermissions(prisonUserMock, PrisonerMockDataA)
+
+        expect(permissions.category).toEqual({
+          view: false,
+          edit: false,
+        })
+      })
+
+      it('should return edit true if user has ReleaseDatesCalculator role', async () => {
+        const user = { ...prisonUserMock, userRoles: [Role.ReleaseDatesCalculator] }
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.calculateReleaseDates).toEqual({
+          view: false,
+          edit: true,
+        })
+      })
+    })
+
+    describe('caseNotes', () => {
```

Comment:
I think I perhaps preferred the original parameterised test style for this, think I find it easier to read: 

https://github.com/ministryofjustice/hmpps-prisoner-profile/pull/658/files#diff-f2abab7599c1a00864d045e892ea14706179469fe03671a1ef4e8545fcc891faL39-L50

But that's just personal preference, so happy to let it slide, it looks like all the cases are covered